app/components/progress_bar_component/progress_bar_component.html.erb

```diff
@@ -0,0 +1,21 @@
+<div class="govuk-width-container">
+  <div class="hero__heading-wrapper hero__heading-wrapper--full-width govuk-body govuk-!-margin-top-3"> 
+    <h4><strong>Your certificate progress</strong></h4>
+    <p class="govuk-body-s ncce-progress-section--description">
+      <strong>Complete the below training and activities to receive your <%= @programme.title %> certificate.
+      You can complete these at your own pace and in any order.</strong>
+    </p>
+    <p class="govuk-body-s">Choose your first step to accreditation:</p>
+    <div class="ncce-progress-section">
+      <% PROG_TITLE_TO_GROUPINGS[@programme.title].each_with_index do |grouping_title, index| %>
+        <% class_name = @class_names["class_name_#{index + 1}"] %>
+        <a href="#" class="<%= class_name %> ncce-link ncce-progress-section--text">
+          <%= grouping_title %>
+        </a>
+        <% if index < PROG_TITLE_TO_GROUPINGS[@programme.title].size - 1 %>
+          <div class="icon-dotted-line"></div>
+        <% end %>
+      <% end %>
```

Comment:
It is really useful for later maintenance to keep template logic simple. Try to pass the simplest collection possible from the Ruby file to the template. You can calculate the CSS class here (as in this suggestion) or in the controller: either way the extra simplicity will be well worth the effort. Use this suggestion as a guide:

```suggestion
      <% @groupings.each_with_index do |grouping, index| %>
        <% class_name = grouping&.user_complete?(current_user) ? "icon-ticked-circle" : "icon-blank-circle"] %>
        <a href="#" class="<%= class_name %> ncce-link ncce-progress-section--text">
          <%= grouping.title %>
        </a>
        <% if index < (@groupings.size - 1) %>
          <div class="icon-dotted-line"></div>
        <% end %>
      <% end %>
```
Also there is a handy trick in Rails to avoid using indexes to figure out where to put separators such as the dotted line icon: have a look at spacer templates: https://guides.rubyonrails.org/layouts_and_rendering.html#spacer-templates

Reply:
refactored this too thank you